### PR TITLE
use IP instead of hostname

### PR DESCRIPTION
**Pull request by Technoboy-**
when in docker env, hostname is not working

### CODE REVIEW

File: uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.Collection;
+import java.util.Enumeration;
+
+
+/**
+ * @author Tboy
+ */
+public class NetUtils {
+
+  public static String getFirstNoLoopbackIP4Address() {
```
Comment (yangy0000) [inline uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java]: what's the problem this pr aims to resolve? please provide more details 
also, please provide test plan

File: uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    Collection<String> allNoLoopbackIP4Addresses = getNoLoopbackIP4Addresses();
+    if (allNoLoopbackIP4Addresses.isEmpty()) {
+      return null;
+    }
+    return allNoLoopbackIP4Addresses.iterator().next();
+  }
+
+  public static Collection<String> getNoLoopbackIP4Addresses() {
```
Comment (yangy0000) [inline uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java]: please add comments for this method

File: uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java (hunk 1 above)
Comment (Technoboy-) [reply uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java]: if we using uReplicator in federated mode, controller has to register itself using hostname. this is ok when we in physical env.  but if in docker, hostname maybe like "ureplicator-manager-5b7d66988d-9j7hk ",  rest service is not working anymore。

**Comment by Technoboy-** [comment]
for rolling restart worker,  partition assigning may result in replicating crash.

**Pull request closed without merging**